### PR TITLE
feat(stdlib): InfluxDB sample data package

### PR DESCRIPTION
**Pull request by sanderson**
Add an InfluxDB sample data package to easily load InfluxDB sample data sets. The package is part of the `influxdata/influxdb` package and contains two functions:

**sample.data**: Download and output a sample dataset.
**sample.list**: List available sample datasets.

```js
import "influxdata/influxdb/sample"

sample.list()
sample.data(set: "airSensor")
```

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/influxdata/influxdb/sample/sample.flux [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+//
+data = (set) => {
+    setInfo = _setInfo(set: set)
+
+    url = if setInfo.url == "" then
+        die(msg: "Invalid sample data set. Use sample.list to view available datasets.")
+    else
+        setInfo.url
+
+    return csv.from(url)
```
Comment (nathanielc) [inline stdlib/influxdata/influxdb/sample/sample.flux:58]: nit, but we prefer the long form syntax in the stdlib of pasisng parameters, `csv.from(url: url)` could you update this?

File: stdlib/influxdata/influxdb/sample/sample.flux [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+//    - **usgs**: USGS earthquake data from the last week.
+//
+// ## Load InfluxDB sample data
+//
+// ```
+// import "influxdata/influxdb/sample"
+//
+// sample.data(set: "airSensor")
+// ```
+//
```
Comment (nathanielc) [inline stdlib/influxdata/influxdb/sample/sample.flux:49]: I love these doc comments !

File: stdlib/influxdata/influxdb/sample/list_test.flux [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+expected = "
+#group,false,false,false
+#datatype,string,long,boolean
+#default,_result,,
+,result,table,_value
+,,0,true
+"
+
+t_list = () => {
```
Comment (nathanielc) [inline stdlib/influxdata/influxdb/sample/list_test.flux:16]: nit, we have stopped creating functions just for the test case. The new best practice is to simply put the logic directly into the testcase block.

```
testcase list {
    got = sample.list()
        |> count()
        |> map(fn: (r) => ({_value: r._value > 0}))
    want = csv.from(csv: expected)

    testing.diff(got: got, want: want)
}
```

Could you update this?

**Comment by nathanielc** [comment]
Also FWIW @sanderson and I chatted and discussed that using github to host the content will not work as there are rate limits. Work is underway to host this data on a more permanent provider. Once that is done this should be good to go.

**Comment by russorat** [comment]
The following datasets are available now:

https://influx-testdata.s3.amazonaws.com/usgs-earthquake-all-week-annotated.csv
https://influx-testdata.s3.amazonaws.com/noaa-ndbc-latest-observations-annotated.csv
https://influx-testdata.s3.amazonaws.com/air-sensor-data-annotated.csv

**Review by nathanielc — APPROVED** [review]
LGTM Thanks for making the changes

**Pull request merged**